docs/development.md

```diff
@@ -28,44 +30,120 @@ Plug-ins listed under the `plugins` array are then loaded and interrogated for k
 
 Express waits for a resolved configuration file before starting the server.
 
-## Running locally
+## MongoDB
 
-To run the server locally, first install its dependencies:
+Indiekit uses a MongoDB database for persistence. A convenient way to run MongoDB locally is to use [Docker Compose](https://docs.docker.com/compose/). Since the filesystem of a Docker container is ephemeral, you will need to create a [Docker volume](https://docs.docker.com/storage/volumes/) first.
 
 ```sh
-npm install
+docker volume create mongo-data
+```
+
+Create a `docker-compose.yml` file that runs a service on an available port of your Docker host (e.g. 27018) and uses the Docker volume you have just created.
+
+```yml
+version: '3.9'
+services:
+  mongo:
+    container_name: mongo
+    environment:
+      - MONGO_INITDB_ROOT_USERNAME
+      - MONGO_INITDB_ROOT_PASSWORD
+    image: mongo:7.0.11
+    network_mode: bridge
+    ports:
+    - '27018:27017'
+    restart: always
+    volumes:
+      - mongo-data:/data/db
+volumes:
+  mongo-data:
+    external: true
+```
+
+This particular configuration tells Docker to run the `mongo` service on port 27017 of the `mongo` container, and exposes port 27018 on the Docker host (e.g. your laptop).
+
+You can set the necessary environment variables in a `.env` file, or use an `.envrc` file that will be automatically loaded by [direnv](https://direnv.net/) when
+you enter the root of the project.
+
+```txt
+export MONGO_INITDB_ROOT_USERNAME="user"
+export MONGO_INITDB_ROOT_PASSWORD="password"
+export MONGO_URL="mongodb://$MONGO_INITDB_ROOT_USERNAME:$MONGO_INITDB_ROOT_PASSWORD@localhost:27018"
+
 ```
 
-The provided configuration file allows some options to be assigned using environment variables.
+> [!TIP]
+> If you want to inspect data stored in a MongoDB database, you can use the [MongoDB shell](https://www.mongodb.com/products/tools/shell) or a GUI like [Compass](https://www.mongodb.com/products/tools/compass).
+
+## Configure the content store
+
+Indiekit performs CRUD operations on files. These files needs to be stored in a so-called content store. Different content stores require different configurations and credentials.
+
+### GitHub content store
+
+If you want to use a GitHub repository as your content store, you will need to provide a [GitHub personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens). If you create a [fine-grained personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#fine-grained-personal-access-tokens), give it **Read and write** access to **Contents**, and set a reasonable expiration (e.g. 90 days).
+
+```txt
+export GITHUB_USER="johndoe"
+export GITHUB_REPO="indiekit-content"
+export GITHUB_BRANCH="main"
+export GITHUB_TOKEN="github_pat_*****"
+```
+
+## Configure the syndicators
```

Comment:
I think this can be an optional step, as only relevant if you want to test syndicating posts.

```suggestion
## Configure a syndicator (optional)
```